### PR TITLE
fix: handle LS relationships + handle dashboard

### PR DESCRIPTION
**Pull request by Hardikl**
LS(load sharing) relationships don't have `relationship_id` field in ZAPI/REST response, due to this, we either skip them or wrongly populate in Harvest.

Changes made:
- added instancekey as `relationship_id`+`destination_path`
- make sure ZAPI and REST collector would parse as expected
- make sure no impact in dashboard due to this

With this, We can now monitor LS relationships as well and can show in dashboard if required.

### CODE REVIEW

File: cmd/collectors/rest/rest.go [hunk 1]
```diff
@@ -421,8 +421,7 @@ func (r *Rest) HandleResults(result []gjson.Result, prop *prop, isEndPoint bool)
 			if value.Exists() {
 				instanceKey += value.String()
 			} else {
-				r.Logger.Warn().Str("key", k).Msg("skip instance, missing key")
-				break
```
Comment (Hardikl) [inline cmd/collectors/rest/rest.go:425]: Intention here to make behaviour same as ZAPI.

**Review by cgrinds — APPROVED** [review]

File: cmd/collectors/rest/rest.go [hunk 2]
```diff
@@ -421,8 +421,7 @@ func (r *Rest) HandleResults(result []gjson.Result, prop *prop, isEndPoint bool)
...
+				r.Logger.Warn().Str("key", k).Msg("missing key")
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/rest/rest.go:424]: nope.

I we do that, all cluster_scoped one won't created

[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/rest/rest.go]: done

File: cmd/collectors/restperf/restperf.go [hunk 3]
```diff
@@ -504,14 +504,20 @@ func (r *RestPerf) PollData() (map[string]*matrix.Matrix, error) {
...
 
-			// extract instance key(s)
-			for _, k := range instanceKeys {
-				value := parseProperties(instanceData, k)
-				if value.Exists() {
-					instanceKey += value.String()
-				} else {
-					r.Logger.Warn().Str("key", k).Msg("skip instance, missing key")
-					break
+			if len(instanceKeys) != 0 {
```
Comment (cgrinds) [inline cmd/collectors/restperf/restperf.go:507]: is this guard clause needed? range handles empty and nil slice

Comment (Hardikl) [reply cmd/collectors/restperf/restperf.go:507]: Agreed that range would handle nil and empty. But here, we need to isolate these 2 cases,
1. instancekeys not specified --> `cluster scoped templates` --> here we create instance with empty("") instanceKey and proceed as expected.
2. instancekeys specified but got empty response from ONTAP --> here we log and skip that instance.

**Pull request merged**